intellij-plugin-structure/structure-base/src/main/kotlin/com/jetbrains/plugin/structure/base/problems/InvalidDescriptorErrors.kt

```diff
@@ -4,127 +4,135 @@
 
 package com.jetbrains.plugin.structure.base.problems
 
-import com.jetbrains.plugin.structure.base.plugin.PluginProblem
-import com.jetbrains.plugin.structure.base.plugin.ProblemSolutionHint
 
 /**
  * Indicates an issue with plugin descriptor (`plugin.xml`).
  *
  * Such kinds of errors are treated in a special way.
- * Although they are indicated as [errors][com.jetbrains.plugin.structure.base.plugin.PluginProblem.Level.ERROR],
+ * Although they are indicated as [errors][PluginProblem.Level.ERROR],
  * they do not prevent successful creation of plugin by [com.jetbrains.plugin.structure.intellij.plugin.IdePluginManager].
  */
-abstract class InvalidDescriptorProblem(private val descriptorPath: String?) : PluginProblem() {
-  abstract val detailedMessage: String
-
-  override val message
-    get() = "Invalid plugin descriptor" + (if (descriptorPath.isNullOrEmpty()) ": " else " '$descriptorPath': ") + detailedMessage
+abstract class InvalidDescriptorProblem(
+  private val descriptorPath: String?,
+  private val detailedMessage: String
+) : PluginProblem() {
+  final override val message: String
+    get() = "Invalid plugin descriptor" +
+            (" '$descriptorPath'.".takeIf { descriptorPath.isNullOrBlank().not() } ?: ".") +
+            (" $detailedMessage".takeIf { detailedMessage.isNotBlank() } ?: "")
 }
 
-
-class InvalidPluginIDProblem(private val id: String) : PluginProblem() {
-  override val message
-    get() = "Plugin id contains unsupported symbols: $id."
+class InvalidPluginIDProblem(id: String) : InvalidDescriptorProblem(
+  descriptorPath = "id",
```

Comment:
Yeah, but there are so many changes in this branch, that could case a problems. Let's postpone this change, since it will probably require changes in other errors too and probably some changes in inheritance?